app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryUpdate.kt

```diff
@@ -29,10 +29,26 @@ class CustomDrugEntryUpdate : Update<CustomDrugEntryModel, CustomDrugEntryEvent,
         dispatch(RemoveDrugFromPrescription(update.prescribedDrugUuid))
       }
       is DrugFetched -> drugFetched(model, event.drug)
-      is DrugFrequencyChoiceItemsLoaded -> next(model.drugFrequencyChoiceItemsLoaded(event.drugFrequencyChoiceItems.items))
+      is DrugFrequencyChoiceItemsLoaded -> {
+        drugFrequencyChoiceItemsLoaded(event, model)
+      }
     }
   }
 
+  private fun drugFrequencyChoiceItemsLoaded(
+      event: DrugFrequencyChoiceItemsLoaded,
+      model: CustomDrugEntryModel
+  ): Next<CustomDrugEntryModel, CustomDrugEntryEffect> {
+    val drugFrequencyToFrequencyChoiceItemMap = drugFrequencyToFrequencyChoiceItemMap(event.drugFrequencyChoiceItems.items, model.frequency)
+    return next(model.drugFrequencyChoiceItemsLoaded(event.drugFrequencyChoiceItems.items).drugFrequencyToFrequencyChoiceItemMapLoaded(drugFrequencyToFrequencyChoiceItemMap))
```

Comment:
I don't understand what we are trying to do here 😅 . Can you give me an overview of it, please?

Reply:
I'm mapping the frequency value to the `DrugFrequencyChoiceItem` list Value. So that whenever we need to access these values we can just do a `mapVariable.get[frequency]`. Here, I'm loading mapping the frequency value to the `DrugFrequencyChoiceItem` list element. This is from this [PR](https://github.com/simpledotorg/simple-android/pull/3011#discussion_r698237641) comment.

Reply:
@msasikanth Can you let me know if you need more explanation on this?

Reply:
So would we remove `drugFrequencyChoiceItems` from the model later?

Reply:
Yeah @msasikanth